src/main/java/org/passay/AbstractMessageResolver.java

```diff
@@ -29,15 +56,49 @@ public String resolve(final RuleResultDetail detail)
     for (String key : detail.getErrorCodes()) {
       message = getMessage(key);
       if (message != null) {
-        return String.format(message, detail.getValues());
+        final String format;
+        if (locale != null) {
+          format = String.format(locale, message, resolveMatchBehavior(detail.getValues()));
+        } else {
+          format = String.format(message, resolveMatchBehavior(detail.getValues()));
+        }
+        return format;
       }
     }
     final String format;
     if (!detail.getParameters().isEmpty()) {
-      format = String.format("%s:%s", detail.getErrorCode(), detail.getParameters());
+      if (locale != null) {
+        format =  String.format(locale, "%s:%s", detail.getErrorCode(), detail.getParameters()).toString();
+      } else {
+        format =  String.format("%s:%s", detail.getErrorCode(), detail.getParameters()).toString();
+      }
     } else {
       format = detail.getErrorCode();
     }
     return format;
   }
+
+
+  /**
+   * Resolves any custom messaging for {@link MatchBehavior}. The supplied array may be modified by this method.
+   *
+   * @param  args  containing any MatchBehavior enums
+   *
+   * @return  resolved args if custom match behavior messaging was found
+   */
+  protected Object[] resolveMatchBehavior(final Object[] args)
+  {
+    if (args != null && args.length > 0) {
+      for (int i = 0; i < args.length; i++) {
+        final Object arg = args[i];
+        if (arg instanceof MatchBehavior) {
+          final String msg = getMessage(((MatchBehavior) arg).upperSnakeName());
+          if (msg != null) {
+            args[i] = msg;
```

Comment:
The type change here from `MatchBehavior` to `String` raises an eyebrow. It all works because these are just fed into a format string and both types can be converted to strings, but this is brittle and could introduce runtime bugs under other usage patterns.

Reply:
Removed this code in the latest commit.